### ISSUE TITLE
Add Android Auto Messaging support 

### ISSUE THREAD

**Issue by snaggen** [WhisperSystems/Signal-Android#5619]
It would be nice if Signal would work with Android Auto. From what I understand it doesn't seem to be that complicated. It seems to need a notification, and handling a Intent for replies. Not sure if this is the kind of notifications and Intents already supported... in that case it might just be a manifest change... 

https://developer.android.com/training/auto/messaging/index.html

**Comment by Norbert80** [comment]
Please not. I use Signal because it has an encrypted data base in addition to e2e encrypted transport.
I don't want messages leaked to unsecure infotainment systems.

**Comment by snaggen** [comment]
With android auto, the applications are still running on your phone (which is connected by USB cable), and the infotainment system basically acts as a secondary screen for the phone.

**Comment by Norbert80** [comment]
... that maybe captured.

**Comment by snaggen** [comment]
Yes, but from what I know, there are more viruses and trojans on Android Phones than on infotainment systems. Also, there are more viruses and trojans on desktop computers, still there is a Signal desktop client. If you don't trust Android Auto, you don't need to use it.... I really don't see the problem.

**Comment by Norbert80** [comment]
This would give others the possibility to leak MY messages.

Go and use a mainstream messenger.

**Comment by snaggen** [comment]
Yes, and if you are a person in need of that level of security, you shouldn't use Android Auto. Hence, this would not be a problem. You understand that just because Android Auto support would be added, no one forces you to use it. Just don't connect the cable to your phone. You are free to choose for your self, based on your security requirements.

**Comment by breversa** [comment]
@snaggen 
What @Norbert80 means is that if HE doesn't use Android Auto, but messages you (for instance) and YOU are using Android Auto, your conversation (that is HIS and YOUR messages) may be compromised from your system.

**Comment by snaggen** [comment]
@breversa @Norbert80 , that is true and a valid point. However, that is already the issue. You don't know what I have installed on my phone. I already leak all the messages, through the notifications, to various applications (like Pebble, Android Ware and others). Peoples phone setup is probably more insecure than a car infotainment system. And if I have notification sync applications like PUshbullet installed, the situation would be even worse. I don't even dare to analyze the Signal for the desktop chrome app.... 

In short, you are already forced to trust my setup, and I don't see how Android Auto would make that any worse than it already is....

**Comment by snaggen** [comment]
@Norbert80 @breversa to address the risk of the recipient leaking my messages, maybe there should be a per conversation setting to not include any message text in the notifications. So that I can mark the conversation as "top secret" and know that the notifications on the other side will not send my message to a third party. Instead the notificaiton would only say something generic like "You have a new message". But that should be its on feature request I guess....

**Comment by breversa** [comment]
I was simply trying to clear out what @Norbert80 meant. I haven't thought into the matter itself myself, but as you point out, it's already an issue.

**Comment by Norbert80** [comment]
@snaggen I really can't understand these "arguments". What you say is that because there is already risk of messages being leaked it doesn't matte to implement additional risks...

**Comment by moxie0** [comment]
If someone submits a PR for android auto support I would accept it.
